tests/testthat/test-helpers_design.R

```diff
@@ -205,3 +205,203 @@ test_that("h_determine_dlts returns correctly updated data object for placebo =
   expect_s4_class(result, "Data")
   expect_equal(result, expected_result)
 })
+
+test_that("simulate for the class design works end to end", {
+  emptydata <- Data(doseGrid = c(1, 3, 5, 10, 15, 20, 25, 40, 50, 80, 100))
+
+  model <- LogisticLogNormal(
+    mean = c(-0.85, 1),
+    cov =
+      matrix(c(1, -0.5, -0.5, 1),
+        nrow = 2
+      ),
+    ref_dose = 56
+  )
+
+  myNextBest <- NextBestNCRM(
+    target = c(0.2, 0.35),
+    overdose = c(0.35, 1),
+    max_overdose_prob = 0.25
+  )
+
+  mySize1 <- CohortSizeRange(
+    intervals = c(0, 30),
+    cohort_size = c(1, 3)
+  )
+  mySize2 <- CohortSizeDLT(
+    intervals = c(0, 1),
+    cohort_size = c(1, 3)
+  )
+  mySize <- maxSize(mySize1, mySize2)
+
+  myStopping1 <- StoppingMinCohorts(nCohorts = 3)
+  myStopping2 <- StoppingTargetProb(
+    target = c(0.2, 0.35),
+    prob = 0.5
+  )
+  myStopping3 <- StoppingMinPatients(nPatients = 20)
+  myStopping <- (myStopping1 & myStopping2) | myStopping3
+
+  myIncrements <- IncrementsRelative(
+    intervals = c(0, 20),
+    increments = c(1, 0.33)
+  )
+
+  design <- Design(
+    model = model,
+    nextBest = myNextBest,
+    stopping = myStopping,
+    increments = myIncrements,
+    cohort_size = mySize,
+    data = emptydata,
+    startingDose = 3
+  ) # check if default design constructor is enough to test simulate method
+
+  myTruth <- probFunction(model, alpha0 = 7, alpha1 = 8)
+
+  options <- McmcOptions(
+    burnin = 100,
+    step = 2,
+    samples = 1000,
+    rng_kind = "Mersenne-Twister",
+    rng_seed = 1234
+  )
+  time <- system.time(mySims <- simulate(design,
```

Comment:
do we need to time this simulation run? (not sure, because `time` is also not used below and would anyway differ between platforms)

Reply:
fully agree - was a bit lazy from me taking that over from the examples...